src/overlays/actors/ovl_En_Vm/z_en_vm.c

```diff
@@ -27,35 +36,497 @@ const ActorInit En_Vm_InitVars = {
+    { { { 0.0f, 0.0f, 0.0f }, { 0.0f, 0.0f, 0.0f }, { 0.0f, 0.0f, 0.0f }, { 0.0f, 0.0f, 0.0f } } },
+};
+
+static Vec3f D_80B2EAEC = { 0.0f, 0.0f, 0.0f };
+
+static Vec3f D_80B2EAF8 = { 0.0f, 0.0f, 0.0f };
+
+static Vec3f D_80B2EB04 = { 500.0f, 0.0f, 0.0f };
+
+static Vec3f D_80B2EB10 = { -500.0f, 0.0f, 0.0f };
+
+static Vec3f D_80B2EB1C = { 0.0f, 0.0f, 0.0f };
+
+static Vec3f D_80B2EB28 = { 0.0f, 0.0f, 1600.0f };
+
+static Vec3f D_80B2EB34 = { 1000.0f, 700.0f, 2000.0f };
+
+static Vec3f D_80B2EB40 = { 1000.0f, -700.0f, 2000.0f };
+
+static Vec3f D_80B2EB4C = { -1000.0f, 700.0f, 1500.0f };
+
+static Vec3f D_80B2EB58 = { -1000.0f, -700.0f, 1500.0f };
+
+static Vec3f D_80B2EB64 = { 500.0f, 0.0f, 0.0f };
+
+static Vec3f D_80B2EB70 = { -500.0f, 0.0f, 0.0f };
+
+static Vec3f D_80B2EB7C = { 0.4f, 0.4f, 0.4f };
+
+static UNK_PTR D_80B2EB88[] = { 0x0402CFE0, 0x0402D7E0, 0x0402DFE0, 0x0402E7E0, 0x0402EFE0,
+                                0x0402F7E0, 0x0402FFE0, 0x040307E0, 0x04030FE0, 0x040317E0 };
+
+extern SkeletonHeader D_06003F60;
+extern AnimationHeader D_06000068;
+extern Gfx D_04031FE0[];
+extern Gfx D_06002728[];
+
+void EnVm_SetupAction(EnVm* this, EnVmActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void EnVm_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnVm* this = THIS;
+
+    SkelAnime_Init(globalCtx, &this->skelAnime, &D_06003F60, &D_06000068, this->limbDrawTable,
+                   this->transitionDrawTable, 11);
+    ActorShape_Init(&thisx->shape, 0.0f, NULL, 0.0f);
+    Collider_InitCylinder(globalCtx, &this->colliderCylinder);
+    Collider_SetCylinder(globalCtx, &this->colliderCylinder, thisx, &sCylinderInit);
+    Collider_InitQuad(globalCtx, &this->colliderQuad1);
+    Collider_SetQuad(globalCtx, &this->colliderQuad1, thisx, &sQuadInit1);
+    Collider_InitQuad(globalCtx, &this->colliderQuad2);
+    Collider_SetQuad(globalCtx, &this->colliderQuad2, thisx, &sQuadInit2);
+    this->beamSightRange = (thisx->params >> 8) * 40.0f;
+    thisx->params &= 0xFF;
+    thisx->naviEnemyId = 0x39;
+
+    if (thisx->params == BMT_LARGE) {
+        thisx->colChkInfo.health = 2;
+        Actor_SetScale(thisx, 0.014f);
+    } else {
+        thisx->colChkInfo.health = 1;
+        Actor_SetScale(thisx, 0.01f);
+    }
+
+    EnVm_SetupWait(thisx);
+}
+
+void EnVm_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnVm* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->colliderCylinder);
+}
+
+void EnVm_SetupWait(EnVm* this) {
+    f32 frameCount = SkelAnime_GetFrameCount(&D_06000068);
+
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06000068, 1.0f, frameCount, frameCount, 2, 0.0f);
+    this->unk_25E = this->unk_260 = 0;
+    this->unk_21C = 0;
+    this->timer = 10;
+    EnVm_SetupAction(this, EnVm_Wait);
+}
+
+void EnVm_Wait(EnVm* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    f32 fVar;
+    s16 sVar;
+    s16 pad;
```

Comment:
You should be able to replace 2 s16 pads with 1 s32 pad

Reply:
I guess not necessarily, but it's worth a try

Reply:
The "sVar" isn't padding iirc but I should probably rename these variables anyway

Reply:
Oh nevermind I see what you mean now, using an s32 does not work though for some reason.

Reply:
I was referring to pad and pad1

Reply:
Yeah, making a single s32 pad does not match